### PR TITLE
Update crate description for `identity-core`

### PR DESCRIPTION
**Pull request by olivereanderson**
# Description of change
Updates the crate description for `identity-core`. Also fixes all reported warnings during documentation generation of this crate. 

## Links to any relevant issues
Necessary for https://github.com/iotaledger/identity.rs/issues/133

## Type of change
Add an `x` to the boxes that are relevant to your changes.

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] Enhancement (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation Fix

## How the change has been tested
Runs locally without warnings under `./identity-core/$ cargo doc --no-deps --open`. 

## Change checklist
Add an `x` to the boxes that are relevant to your changes.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: identity-core/README.md [hunk 1]
```diff
@@ -1,5 +1,26 @@
...
+- [`Timestamp`](crate::common::Timestamp): Representing an [RFC3339 compliant](https://datatracker.ietf.org/doc/html/rfc3339) timestamp. 
+- [`Url`](crate::common::Url): Representing a Url. 
+- More collections such as [`OneOrMany`](crate::common::OneOrMany), [`OneOrSet`](crate::common::OneOrSet) and [`OrderedSet`](crate::common::OrderedSet). 
+
+All of these data types are located in the [`common` module](crate::common).  
+
+## Cryptographic primitives and traits 
+Cryptographic primitives necessary for DID related operations such as signing and verifying data can be found in the [`crypto` module](crate::crypto). 
+
+Although the IOTA Identity Framework strives towards keeping cryptographic primitives and operations as internal implementation details, it does also expose some lower level apis requiring more interaction with this module. The latter setting will typically involve the [`KeyPar`](crate::crypto::KeyPair) type representing a cryptographic key pair.
```
Comment (eike-hass) [inline identity-core/README.md:21]:
```suggestion
Although the IOTA Identity Framework strives towards keeping cryptographic primitives and operations as internal implementation details, it does also expose some lower level apis requiring more interaction with this module. The latter setting will typically involve the [`KeyPair`](crate::crypto::KeyPair) type representing a cryptographic key pair. 
```

File: identity-core/README.md [hunk 2]
```diff
@@ -1,5 +1,26 @@
...
+
+All of these data types are located in the [`common` module](crate::common).  
+
+## Cryptographic primitives and traits 
+Cryptographic primitives necessary for DID related operations such as signing and verifying data can be found in the [`crypto` module](crate::crypto). 
+
+Although the IOTA Identity Framework strives towards keeping cryptographic primitives and operations as internal implementation details, it does also expose some lower level apis requiring more interaction with this module. The latter setting will typically involve the [`KeyPair`](crate::crypto::KeyPair) type representing a cryptographic key pair. 
+
+## Convenient JSON serialization
+By bringing the [`ToJson`](crate::convert::ToJson) and [`FromJson`](crate::convert::FromJson) traits from this crate into scope, one can conveniently convert most types from the IOTA Identity Framework to and from a few common JSON representations.
```
Comment (cycraig) [inline identity-core/README.md:24]: ## Convenient JSON Serialization
The [`ToJson`](crate::convert::ToJson) and [`FromJson`](crate::convert::FromJson) traits from this crate provide convenience functions to convert most types from the IOTA Identity Framework to and from a few common JSON representations.

File: identity-core/README.md [hunk 3]
```diff
@@ -1,5 +1,26 @@
...
+
+## Common data types 
+This crate introduces data types necessary for many concepts within the IOTA Identity Framework. These types include 
+- [`Context`](crate::common::Context): Representing [JSON-LD contexts](https://www.w3.org/TR/vc-data-model/#contexts)
+- [`Fragment`](crate::common::Fragment): Representing a [DID Url fragment](https://www.w3.org/TR/did-core/#dfn-did-fragments)
+- [`Timestamp`](crate::common::Timestamp): Representing an [RFC3339 compliant](https://datatracker.ietf.org/doc/html/rfc3339) timestamp. 
+- [`Url`](crate::common::Url): Representing a Url. 
+- More collections such as [`OneOrMany`](crate::common::OneOrMany), [`OneOrSet`](crate::common::OneOrSet) and [`OrderedSet`](crate::common::OrderedSet). 
+
+All of these data types are located in the [`common` module](crate::common).
```
Comment (cycraig) [inline identity-core/README.md:16]:
```suggestion
## Common Data Types 
This crate exposes some fundamental data types used across the IOTA Identity Framework:
- [`Context`](crate::common::Context): represents [JSON-LD contexts](https://www.w3.org/TR/vc-data-model/#contexts).
- [`Fragment`](crate::common::Fragment): a [DID URL fragment](https://www.w3.org/TR/did-core/#dfn-did-fragments).
- [`Timestamp`](crate::common::Timestamp): an [RFC3339](https://datatracker.ietf.org/doc/html/rfc3339) compliant timestamp.
- [`Url`](crate::common::Url): a URL string. 
- Collections: [`OneOrMany`](crate::common::OneOrMany), [`OneOrSet`](crate::common::OneOrSet), [`OrderedSet`](crate::common::OrderedSet). 

The above types are located in the [`common` module](crate::common).  
```

File: identity-core/README.md [hunk 4]
```diff
@@ -1,5 +1,26 @@
...
+- [`Timestamp`](crate::common::Timestamp): Representing an [RFC3339 compliant](https://datatracker.ietf.org/doc/html/rfc3339) timestamp. 
+- [`Url`](crate::common::Url): Representing a Url. 
+- More collections such as [`OneOrMany`](crate::common::OneOrMany), [`OneOrSet`](crate::common::OneOrSet) and [`OrderedSet`](crate::common::OrderedSet). 
+
+All of these data types are located in the [`common` module](crate::common).  
+
+## Cryptographic primitives and traits 
+Cryptographic primitives necessary for DID related operations such as signing and verifying data can be found in the [`crypto` module](crate::crypto). 
+
+Although the IOTA Identity Framework strives towards keeping cryptographic primitives and operations as internal implementation details, it does also expose some lower level apis requiring more interaction with this module. The latter setting will typically involve the [`KeyPair`](crate::crypto::KeyPair) type representing a cryptographic key pair.
```
[the remark this answers is not in the document]
Comment (olivereanderson) [reply identity-core/README.md]: You are exceptionally good at clean and concise explanations @cycraig!

[the remark this answers is not in the document]
Comment (cycraig) [reply identity-core/README.md]: Only when editing someone else's work, I'm rubbish at writing... sorry for all the spam.

File: identity-core/README.md [hunk 5]
```diff
@@ -1,5 +1,26 @@
...
 The core types and utilities for IOTA identity.
 
-This crate only contains generic types and utilities for use across the other identity crates such
-as. See the `identity_did`, `identity_iota`, and `identity_credential` crates for DID-specific
-types and utilities.
+The types and traits exposed by this crate are mainly intended to be used by the other crates constituting the [IOTA Identity Framework](https://wiki.iota.org/identity.rs/introduction). 
+
+## Common Data Types 
+This crate exposes some fundamental data types used across the IOTA Identity Framework:
+- [`Context`](crate::common::Context): represents [JSON-LD contexts](https://www.w3.org/TR/vc-data-model/#contexts).
```
Comment (abdulmth) [inline identity-core/README.md:10]: Do these links work? I'm not sure if md can resolve rust paths.

**Review by abdulmth — APPROVED** [review]

**Pull request merged**